plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx

```diff
@@ -226,3 +228,53 @@ export const withUpgradeNoticeControls =
 	};
 
 addFilter( 'editor.BlockEdit', metadata.name, withUpgradeNoticeControls );
+
+export const withHandlePreviewState =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
+	( props: BlockEditProps< ProductCollectionAttributes > ) => {
+		if ( ! isProductCollection( props.name ) ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		let previewMessage = __(
+			'Actual products will vary depending on the currently viewed page.',
+			'woocommerce'
+		);
+
+		/**
+		 * Example:
+		 * How to change the preview message based on the collection selected.
+		 */
+		if ( props.attributes.collection === CoreCollectionNames.ON_SALE ) {
+			previewMessage = __(
+				'Custom tooltip for on sale collection.',
+				'woocommerce'
+			);
+		}
+
+		/**
+		 * 3PDs will be able to pass handlePreviewState as a prop & then update the state
+		 * based on their own logic.
+		 * Example: As you can see below, we are updating the state after 5 seconds.
+		 *          Also, we are toggling the isPreview state in each 5 seconds.
+		 */
+		const handlePreviewState = (
+			previewState: PreviewState,
+			setPreviewState: React.Dispatch<
+				React.SetStateAction< PreviewState >
+			>
+		) => {
+			setTimeout( () => {
+				setPreviewState( {
+					isPreview: ! previewState.isPreview,
+					previewMessage,
+				} );
+			}, 5000 );
+		};
+
+		return (
+			// Here we are passing the handlePreviewState as a prop to the BlockEdit.
+			<BlockEdit { ...props } handlePreviewState={ handlePreviewState } />
+		);
+	};
```

Comment:
If you just use props this would become:

```suggestion
       const [preview, setPreview] = useState(true);
       const previewMessage = 'whatever';
       useEffect(() => {
          const timeoutId = setTimeout(() => {
             setPreview(!preview);
          }, 5000);
           return () => clearTimeout(timeoutId);
       }, []);
       
       return (
			<BlockEdit { ...props } previewMessage={previewMessage} isPreview={preview} />
	  );
};
```

Reply:
Just left a comment at https://github.com/woocommerce/woocommerce/pull/45703#discussion_r1535265174. Please let me know if you have any follow-up questions. 🙂